lib/Config/Populator.php

```diff
@@ -549,6 +551,13 @@ private function scheduleAuthorizedSendingEmailsCheck() {
     );
   }
 
+  private function initLastAnnouncementDate() {
+    if (!$this->settings->get('last_announcement_date')) {
+      $beamer = new Beamer($this->settings, WPFunctions::get());
```

Comment:
Just thinking if we should pass the instance through DI here... What do you think?

Reply:
I can make `$wp` a property and initialize it on the constructor, but in order to properly inject it with DI we need to refactor all classes using Populator which may easily bring a lot of refactoring.
Do you think `$wp` as property is good enough?

Reply:
> in order to properly inject it with DI we need to refactor all classes using Populator

I don't think it is that big problem. AFAIK Populator is only used in Activator and Activator is already configured in DI container. So this is not going to be a big refactor. 

If we don't keep adding more and more classes to the DI we will never reach the goal to use DI everywhere.